backend/views.py

```diff
@@ -1,8 +1,36 @@
-from canvas_oauth.oauth import get_oauth_token
+import logging
+
+from canvas_oauth.oauth import get_oauth_token, handle_missing_token
+from canvas_oauth.models import CanvasOAuth2Token
+from canvas_oauth.exceptions import CanvasOAuthError, MissingTokenError, InvalidOAuthStateError, InvalidOAuthReturnError
+
+from django.contrib.auth.models import User
 from django.contrib.auth.decorators import login_required
 from django.shortcuts import render
 
+
+logger = logging.getLogger(__name__)
+
 @login_required
 def get_home_template(request):
-    access_token = get_oauth_token(request)
+    try:
+        access_token = get_oauth_token(request)
+    except InvalidOAuthStateError as err:
+        logger.error("InvalidOAuthStateError. Remove invalid refresh_token and prompt for reauthentication.")
+        CanvasOAuth2Token.objects.filter(user=request.user).delete()
+        return handle_missing_token(request)
+    except InvalidOAuthReturnError as err:
+        logger.error("InvalidOAuthReturnError. Remove invalid refresh_token and prompt for reauthentication.")
+        CanvasOAuth2Token.objects.filter(user=request.user).delete()
+        return handle_missing_token(request)
+    except MissingTokenError as err:
+        logger.error("MissingTokenError. Prompt for reauthentication.")
+        return handle_missing_token(request)
```

Comment:
I **think** this `except` is unnecessary, since if the `MissingTokenError` occurs, the Canvas OAuth middleware will handle the redirect to reauthenticate -- see https://github.com/Harvard-University-iCommons/django-canvas-oauth/blob/6a2dcff2474f4b7bc7367f41d70e0bbb377ee6c0/canvas_oauth/oauth.py#L18-L45

Reply:
Removed the MissingTokenError error handling section.